graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -127,34 +129,23 @@
 from .validation import validate_schema_and_query_ast
 
 
-# LocationStackEntry contains the following:
-# - location: Location object corresponding to an inserted MarkLocation block
-# - num_traverses: Int counter for the number of traverses inserted after the last MarkLocation
-#                  (corresponding Location stored in `location`)
-LocationStackEntry = namedtuple("LocationStackEntry", ("location", "num_traverses"))
+class OutputMetadata(NamedTuple):
+    """Metadata about a query's outputs.
 
+    Containing the following members:
+        type: GraphQL type object, such as GraphQLString or GraphQLInt, describing the type of the
+              output value.
+        optional: boolean, whether the output is part of an optional traversal, which would allow a
+                  value of null.
+        folded: boolean, whether the output is within a fold scope. Note that the "type" is expected
+                to be a GraphQLList unless the output field is an _x_count, in which case the type
+                must be GraphQLInt.
```

Comment:
Thoughts on moving the docs to be inline right next to the field definition itself? That way we won't have to specify the type in the docstring as well as in the type hint (e.g. `folded: boolean` would go away since it would be implied by proximity to the field.